dashboard/modules/job/sdk.py

```diff
@@ -14,13 +15,46 @@
     JOBS_API_ROUTE_SUBMIT, JOBS_API_ROUTE_STOP, JOBS_API_ROUTE_STATUS,
     JOBS_API_ROUTE_PACKAGE)
 
+from ray.client_builder import _split_address
+
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
+@dataclasses.dataclass
+class ClusterInfo:
+    address: str
+    cookies: Optional[Dict[str, Any]]
+    metadata: Optional[Dict[str, Any]]
+
+
 class JobSubmissionClient:
-    def __init__(self, address: str):
-        self._address: str = address.rstrip("/")
+    def __init__(self, address: str, create_cluster_if_needed=False):
+        module_string, inner_address = _split_address(address.rstrip("/"))
+
+        if module_string == "http" or module_string == "https":
+            self._address = address
+            self._cookies = None
+            self._default_metadata = {}
+        else:
+            module = importlib.import_module(module_string)
+            try:
+                module = importlib.import_module(module_string)
+            except Exception:
+                raise RuntimeError(
+                    f"Module: {module_string} does not exist.\n"
+                    f"This module was parsed from Address: {address}"
+                ) from None
+            assert "get_job_submission_client_cluster_info" in dir(module), (
+                f"Module: {module_string} does "
+                "not have `get_job_submission_client_cluster_info`.")
+
+            cluster_info = module.get_job_submission_client_cluster_info(
+                inner_address, create_cluster_if_needed)
+            self._address = cluster_info.address
+            self._cookies = cluster_info.cookies
```

Comment:
should we do some validation for this?

Reply:
the `ClusterInfo` object is a dataclass so I think this should be ok